okhttp/src/main/java/io/grpc/transport/okhttp/OkHttpChannelBuilder.java

```diff
@@ -84,18 +83,19 @@ public void close(ExecutorService executor) {
 
   /** Creates a new builder for the given server host and port. */
   public static OkHttpChannelBuilder forAddress(String host, int port) {
-    return new OkHttpChannelBuilder(new InetSocketAddress(host, port), host);
+    return new OkHttpChannelBuilder(host, port);
   }
 
-  private final InetSocketAddress serverAddress;
   private ExecutorService transportExecutor;
-  private String host;
+  private final String host;
+  private final int port;
+  private String authorityHost;
   private SSLSocketFactory sslSocketFactory;
   private ConnectionSpec connectionSpec = DEFAULT_CONNECTION_SPEC;
 
-  private OkHttpChannelBuilder(InetSocketAddress serverAddress, String host) {
-    this.serverAddress = Preconditions.checkNotNull(serverAddress, "serverAddress");
-    this.host = host;
+  private OkHttpChannelBuilder(String host, int port) {
+    this.host = Preconditions.checkNotNull(host);
+    this.port = port;
```

Comment:
Missing:
```java
this.authorityHost = host;
```

Without it, you would get a NullPointerException if you don't overrideHostForAuthority.